internal/controller/persistentvolume/persistentvolume.go

```diff
@@ -161,11 +179,18 @@ func (r ReconcilePersistentVolume) reconcilePV(obj runtime.Object) error {
 		}
 		defer cr.DeleteCredentials()
 
-		err = rbd.RegenerateJournal(imageName, volumeHandler, pool, journalPool, requestName, cr)
+		rbdVolID, err := rbd.RegenerateJournal(imageName, volumeHandler, pool, journalPool, requestName, cr)
 		if err != nil {
 			util.ErrorLogMsg("failed to regenerate journal %s", err)
 			return err
 		}
+		if rbdVolID != volumeHandler {
+			err = r.storeVolumeIDInPV(pv, rbdVolID)
```

Comment:
imo, the storeVolumeIDinPV should be a protected call. Otherwise the other parallel request may see a inconsistency in some scenarios.